openslides/users/serializers.py

```diff
@@ -57,11 +57,19 @@ class Meta:
 
     def validate(self, data):
         """
-        Checks that first_name or last_name is given. Generates the
+        Checks if the given data is empty. Generates the
         username if it is empty.
         """
-        if not (data.get('username') or data.get('first_name') or data.get('last_name')):
-            raise ValidationError({'detail': _('Username, given name and surname can not all be empty.')})
+        # Check if one of the fields is actually filled
+        filled_field_keys = []
+        for field in data:
+            if data.get(field) != '':
+                filled_field_keys.append(field)
+
+        # Check if filled fields match to any of the given fields
+        intersecting_keys = list(set(filled_field_keys) & set(self.fields.keys()))
```

Comment:
This solution does not work it one sends only some minor field in a PUT context.

I propose to do in line 74 (old file): Only if username is empty and it is not a PATCH context, then first validate that at least one of the names is given, then generate the username.

Reply:
Do I understand correctly: Just execute my new batch of code, if we are in a PATCH context, but do the old stuff, if we are in a PUT context?

I tested the PUT context, iE User creation and it works as expected. I don't quiet get why username, first_name or last_name should be mandatory at that point, when one of the fields is sufficient to create a user in first place? Or should these fields be mandatory upon user creation?

Reply:
Sorry but I missed to answer this. My proposal is not to add much code for the PATCH context but to put some old code into an if clause in case of non-PATCH contexts.